addons/payment/models/account_journal.py

```diff
@@ -70,6 +43,26 @@ def _compute_available_payment_method_ids(self):
 
                 journal.available_payment_method_ids = [Command.unlink(payment_method) for payment_method in to_remove]
 
+    def write(self, vals):
```

Comment:
I just realized when commenting on the use of magic numbers, and after reading your question about being unable to raise in `unlink`, that you should rather use the [odoo.api.ondelete decorator](https://www.odoo.com/documentation/15.0/developer/reference/backend/orm.html#odoo.api.ondelete) to handle this case.